File: sdks/csnet/BaSys40.Technologies.oneM2M/Utils/ResultHandling/Message.cs
using System.Threading;

namespace oneM2MClient.Utils.ResultHandling
{
    public class Message : IMessage
    {
        public MessageType MessageType { get; set; }
        public string Text { get; set; }
        public string Code { get; set; }

        public Message(MessageType messageType, string text) : this(messageType, text, null)
        { }
        public Message(MessageType messageType, string text, string code)
        {
            MessageType = messageType;
            Text = text;
            Code = code;
        }


        public override string ToString()
        {
            return string.Format(Thread.CurrentThread.CurrentCulture,"{0} | {1} - {2}", MessageType, Code, Text);
        }
    }
}

File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/vab/backend/server/utils/Message.java
package org.eclipse.basyx.vab.backend.server.utils;

import java.util.HashMap;

/**
 * 
 * @author pschorn
 *
 */
@SuppressWarnings("serial")
public class Message extends HashMap<String, Object> implements IMessage {
	
	public static final String MESSAGETYPE = "messageType";
	public static final String CODE = "code";
	public static final String TEXT = "text";

	public Message(MessageType messageType, String text) {
		
		this(messageType, null, text);
	} 
	
	
	public Message(MessageType messageType, String code, String text) {
		put(MESSAGETYPE, messageType.getId());
		put(CODE, code);
		put(TEXT, text);
	}

	public String getText() {
		return (String) get(TEXT);
	}

	public String getCode() {
		return (String) get(CODE);
	}

	public MessageType getMessageType() {
		return MessageType.getById((int) get(MESSAGETYPE));
	}
	
	public String toString() {

		if (getCode()==null | getCode().isEmpty()){
			return getMessageType().getId() + " | " + getText();
		} else {
			return getMessageType().getId() + " | " + getCode() + " - " + getText();
		}
	}

}
